tests/unit/test_meta.py

```diff
@@ -450,11 +454,20 @@ def test_create_meta_with_app_desktop_key(self):
         section = "Desktop Entry"
         self.assertTrue(section in contents)
         self.assertThat(contents[section].get("Exec"), Equals("my-package %U"))
+        self.assertThat(
+            contents[section].get("Icon"), Equals("${SNAP}/usr/share/app3.png")
+        )
+        self.assertThat(
+            fake_logger.output,
+            Not(Contains(
+                "Icon {} specified in desktop file {} not found "
+                "in prime directory".format("${SNAP}/usr/share/app3.png", "app3.desktop")
+            )),
+        )
 
         snap_yaml = os.path.join("prime", "meta", "snap.yaml")
         self.assertThat(snap_yaml, Not(FileContains("desktop: app1.desktop")))
         self.assertThat(snap_yaml, Not(FileContains("desktop: app2.desktop")))
-        self.assertThat(snap_yaml, Not(FileContains("desktop: app3.desktop")))
```

Comment:
you still need to verify that the desktop key did not make it to snap.yaml, right?

Reply:
oops, I didn't mean to remove that